src/cli/index.js

```diff
@@ -155,11 +155,16 @@ if (command.requireLockfile && !fs.existsSync(path.join(config.cwd, constants.LO
 //
 const run = (): Promise<void> => {
   invariant(command, 'missing command');
-  return command.run(config, reporter, commander, commander.args).then(() => {
+  return command.run(config, reporter, commander, commander.args).then(exitCode => {
     reporter.close();
     if (outputWrapper) {
       reporter.footer(false);
     }
```

Comment:
Throwing an error causes the `yarn-error.log` file to be written and error message stuff to be printed. Am I missing something here?

I did a find all and none of the commands return anything. They all have the signature of `Promise<void>`. I've updated the PR again.